pmlc/tools/pmlc-jit/pmlc-jit.cc

```diff
@@ -15,58 +15,28 @@
 // thrown.
 //===----------------------------------------------------------------------===//
 
-#include <iostream>
 #include <stdexcept>
-#include <string>
 
 #include "mlir/ExecutionEngine/OptUtils.h"
 #include "mlir/Support/JitRunner.h"
 #include "llvm/Support/InitLLVM.h"
 #include "llvm/Support/TargetSelect.h"
+#include "llvm/Support/raw_ostream.h"
 
 #include "pmlc/util/all_dialects.h"
 #include "pmlc/util/all_passes.h"
 
-#define EXPECTED_EXCEPTION_FLG_START_LEN 35
-#define EXPECTED_EXCEPTION_FLAG_START "-expected-runtime-exception-string="
-
 int main(int argc, char **argv) {
   registerAllDialects();
   llvm::InitLLVM y(argc, argv);
   llvm::InitializeNativeTarget();
   llvm::InitializeNativeTargetAsmPrinter();
   mlir::initializeLLVMPasses();
 
-  // Get the expected-runtime-exception-string and remove the parameters
-  // unknown to the tool that is being delegated to.
-  std::string expectedString;
-  int fixedUpCounter = 0;
-
-  for (int i = 0; i < argc; i++) {
-    if (fixedUpCounter != i) {
-      argv[fixedUpCounter] = argv[i];
-    }
-
-    std::string arg = argv[i];
-    if (arg.find(EXPECTED_EXCEPTION_FLAG_START) == 0) {
-      expectedString = arg.substr(EXPECTED_EXCEPTION_FLG_START_LEN);
-      continue;
-    }
-
-    fixedUpCounter++;
-  }
-
-  // Zero up the fixed parameters that have been moved forward.
-  for (int i = fixedUpCounter; i < argc; i++) {
-    argv[i] = nullptr;
-  }
-
   try {
-    return mlir::JitRunnerMain(fixedUpCounter, argv, nullptr);
+    return mlir::JitRunnerMain(argc, argv, nullptr);
   } catch (const std::runtime_error &e) {
```

Comment:
Catch `std::exception` here, since `std::runtime_error` derives from it.

Reply:
Caught std::exception.
The reason I went with the expected exception was that with this implementation there is a possibiolity we get false positive passing tests - the jit can throw a RT exception and the test can fail. We would not know about it, since the exception will be silently "swallowed" - not for this tests, but for all the rest that there is no CHECK/CHECK-NOT check performed by the FileCheck.

Reply:
That's a really good point about silently swallowing exceptions. I could see having a command line flag to toggle between these when an exception is caught.
* Return 0: only print exceptions
* Return non-zero